# Copyright © José Luis Lara Carrascal 2010-2011 <manualinux@yahoo.es> 
# <http://manualinux.freehostia.com> <http://manualinux.exofire.net> <http://manualinux.heliohost.org>
# Este parche se distribuye bajo la Licencia General GNU

diff -ruN briquolo-0.5.7/src/MOGL/MOGL_Image.cpp briquolo-0.5.7a/src/MOGL/MOGL_Image.cpp
--- briquolo-0.5.7/src/MOGL/MOGL_Image.cpp	2006-03-07 21:53:33.000000000 +0100
+++ briquolo-0.5.7a/src/MOGL/MOGL_Image.cpp	2010-07-16 16:27:11.000000000 +0200
@@ -26,6 +26,7 @@
 using namespace std;
 
 #define PNG_BYTES_TO_CHECK 8
+#define png_infopp_NULL (png_infopp)NULL
 
 MOGL_Image::MOGL_Image(bool p_AutoriserTrou)
  : _Image(NULL), _AutoriserTrou(p_AutoriserTrou), _R(0), _G(0), _B(0), _TypeImage(NON_DEFINI)
@@ -235,7 +236,7 @@
   /* Expand grayscale images to the full 8 bits from 1, 2, or 4 bits/pixel */
   if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8)
   {
-    png_set_gray_1_2_4_to_8(png_ptr);
+    png_set_expand_gray_1_2_4_to_8(png_ptr);
   }
   
   /* Expand paletted or RGB images with transparency to full alpha channels
